gradio/utils.py

```diff
@@ -144,6 +150,51 @@ def swap_blocks(self, demo: Blocks):
         self.alert_change()
 
 
+NO_RELOAD = True
+
+
+def _remove_no_reload_codeblocks(file_path: str):
+    """Parse the file, remove the gr.no_reload code blocks, and write the file back to disk.
+
+    Parameters:
+        file_path (str): The path to the file to remove the no_reload code blocks from.
+    """
+
+    with open(file_path) as file:
+        code = file.read()
+
+    tree = ast.parse(code)
+
+    def _is_gr_no_reload(expr: ast.AST) -> bool:
+        """Find with gr.no_reload context managers."""
+        return (
+            isinstance(expr, ast.If)
+            and isinstance(expr.test, ast.Attribute)
+            and isinstance(expr.test.value, ast.Name)
+            and expr.test.value.id == "gr"
+            and expr.test.attr == "NO_RELOAD"
+        )
+
+    # Find the positions of the code blocks to load
+    for node in ast.walk(tree):
+        if _is_gr_no_reload(node):
+            assert isinstance(node, ast.If)  # noqa: S101
+            node.body = [ast.Pass()]
+
+    # convert tree to string
+    code_removed = ast.unparse(tree)
```

Comment:
It looks like `ast.unparse` was added in Python 3.9: https://docs.python.org/3/library/ast.html#ast.unparse

If we absolutely need this method, we could use this third-party implementation: https://github.com/simonpercivall/astunparse

Reply:
Btw we should confirm that the reload tests are using 3.8 (otherwise, should not have passed)

Reply:
Only 7 months until 3.8 EOL 🤩

I think we can manually modify the source code string so will look into it

Reply:
Thanks @freddyaboulton, ping me whenever I can take another pass at the pr

Reply:
@abidlabs this should be good for review now